packages/office-ui-fabric-react/src/components/Coachmark/Coachmark.tsx

```diff
@@ -327,7 +335,47 @@ export class Coachmark extends BaseComponent<ICoachmarkProps, ICoachmarkState> {
   }
 
   public componentWillUnmount(): void {
+    const { preventDismissOnLostFocus } = this.props;
+
     this._events.off(document, 'keydown', this._onKeyDown, true);
+
+    if (!preventDismissOnLostFocus) {
+      this._events.off(document, 'click', this._dismissOnLostFocus, true);
+      this._events.off(document, 'focus', this._dismissOnLostFocus, true);
+    }
+  }
+
+  public dismiss = (ev?: Event | React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>): void => {
+    const { onDismiss } = this.props;
+
+    if (onDismiss) {
+      onDismiss(ev);
+    }
+  };
+
+  private _addListeners(): void {
+    const { preventDismissOnLostFocus } = this.props;
+
+    this._async.setTimeout(() => {
+      this._events.on(document, 'keydown', this._onKeyDown, true);
+
+      if (!preventDismissOnLostFocus) {
```

Comment:
this can mutate. if so, then what @joschect said applies.

Also unclear why you need to do this asynchronously in a `setTimeout` vs a `componentDidMount`.

Also accessing `document` directly might be incorrect; you should be